homeassistant/components/eq3btsmart/entity.py

```diff
@@ -15,3 +32,63 @@ def __init__(self, entry: Eq3ConfigEntry) -> None:
 
         self._eq3_config = entry.runtime_data.eq3_config
         self._thermostat = entry.runtime_data.thermostat
+        self._attr_device_info = DeviceInfo(
+            name=slugify(self._eq3_config.mac_address),
+            manufacturer=MANUFACTURER,
+            model=DEVICE_MODEL,
+            connections={(CONNECTION_BLUETOOTH, self._eq3_config.mac_address)},
+        )
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return a unique ID."""
+
+        return format_mac(self._eq3_config.mac_address) + (
+            "_" + self.name
+            if not isinstance(self.name, UndefinedType) and self.name
+            else ""
+        )
```

Comment:
We should not base our unique id on the name, as I think it would cause the name to change when you change languages.